connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/util/SSLUtils.java

```diff
@@ -64,7 +65,20 @@ public static SslContextFactory createServerSideSslContextFactory(WorkerConfig c
     public static SslContextFactory createClientSideSslContextFactory(WorkerConfig config) {
         Map<String, Object> sslConfigValues = config.valuesWithPrefixAllOrNothing("listeners.https.");
 
-        final SslContextFactory.Client ssl = new SslContextFactory.Client();
+        // Override this method in order to avoid running into
+        // https://github.com/eclipse/jetty.project/issues/4385, which would otherwise cause this to
+        // break when the keystore contains multiple certificates.
+        // The override here matches the bug fix in Jetty for that issue:
+        // https://github.com/eclipse/jetty.project/pull/4404/files#diff-58640db0f8f2cd84b7e653d1c1540913R2188-R2193
+        // TODO: Remove this override when the version of Jetty for the framework is bumped to
+        //       9.4.25 or later
+        final SslContextFactory.Client ssl = new SslContextFactory.Client() {
+            @Override
+            @SuppressWarnings("deprecation")
+            protected X509ExtendedKeyManager newSniX509ExtendedKeyManager(X509ExtendedKeyManager keyManager) {
```

Comment:
I do not think we want SslContextFactory.Client overwrite newSniX509ExtendedKeyManager which is supposed to be Server side. This will change Client behavior.  Could we just use  SslContextFactory.Server instead if require ClientAuth true ?

Reply:
This is the exact fix that Jetty uses in later versions: https://github.com/eclipse/jetty.project/pull/4404/files#diff-58640db0f8f2cd84b7e653d1c1540913R2188-R2193

Reply:
Will this RestClient in Connect follower node also play Server side role to Connect leader node and follower node's hostname need be verified by Connect leader node?

Reply:
No, when SSL is configured on the server, the `SSLUtils::createServerSideSslContextFactory` method is used to create the SSL context factory which should enable hostname verification.
If the leader has to contact a follower it'll instantiate it's own `RestClient` and communicate with the follower's `RestServer` which will be able to perform hostname verification.